apis/common/v1/connection_details.go

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import "k8s.io/apimachinery/pkg/runtime"
+
+// PublishConnectionDetailsTo represents configuration of a connection secret.
+type PublishConnectionDetailsTo struct {
+	// Name is the name of the connection secret
+	Name string `json:"name"`
+
+	// Metadata is secret store specific key/value pairs to be used as metadata
+	// Please note, expected keys will differ for each store type. For example,
+	// it could be "labels" and "annotations" in case of "Kubernetes", but it
+	// would be "tags" for "AWS Secret Manager".
+	// +kubebuilder:pruning:PreserveUnknownFields
+	Metadata runtime.RawExtension `json:"metadata,omitempty"`
+
+	// SecretStoreConfigRef specifies which secret store config should be used
+	// for this ConnectionSecret.
+	// +kubebuilder:default={"name": "default"}
+	SecretStoreConfigRef Reference `json:"configRef,omitempty"`
```

Comment:
Should this be a `*Reference`?

Reply:
Actually, it would probably be more "future proof" to use a typed reference here. We'd like to make ProviderConfigs do this too - see https://github.com/crossplane/crossplane/issues/2255 for the rationale.

Reply:
Sounds reasonable but I couldn't imagine how that could be done: https://github.com/crossplane/crossplane/issues/2255#issuecomment-1042917462

Changed to `*Reference` for now.